tabs/src/components/BeaverHours/Scheduler.tsx

```diff
@@ -0,0 +1,158 @@
+import { createContext, useContext, useState } from "react";
+
+const ScheduleContext = createContext({
+  state: {} as OfficeHoursSchedule,
+  updateSchedule: (newSchedule: OfficeHoursSchedule) => {},
+});
+
+export function Scheduler() {
+  return <ScheduleForm />;
+}
+
+interface DaySchedule {
+  startAt?: string;
+  endAt?: string;
+  repeated?: number;
+}
+
+interface OfficeHoursSchedule {
+  [key: string]: DaySchedule | undefined;
+  Sunday?: DaySchedule;
+  Monday?: DaySchedule;
+  Tuesday?: DaySchedule;
+  Wednesday?: DaySchedule;
+  Thursday?: DaySchedule;
+  Friday?: DaySchedule;
+  Saturday?: DaySchedule;
+}
+
+export function ScheduleForm() {
+  const [state, setSchedule] = useState({
+    Sunday: {} as DaySchedule,
+    Monday: {} as DaySchedule,
+    Tuesday: {} as DaySchedule,
+    Wednesday: {} as DaySchedule,
+    Thursday: {} as DaySchedule,
+    Friday: {} as DaySchedule,
+    Saturday: {} as DaySchedule,
+  } as OfficeHoursSchedule);
+
+  const updateSchedule = (newSchedule: OfficeHoursSchedule) => {
+    setSchedule(newSchedule);
+  };
+  const value = { state, updateSchedule };
+
+  const HandleSubmit = (event: any) => {
+    console.log("A schedule was submitted:\n", state);
+    event.preventDefault();
+  };
+
+  return (
+    <ScheduleContext.Provider value={value}>
+      <form onSubmit={HandleSubmit}>
+        <GenerateWeekTable />
+        <input type="submit" value="submit" />
+      </form>
+    </ScheduleContext.Provider>
+  );
+}
+
+export function GenerateWeekTable() {
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th></th>
+          <th>Start At:</th>
+          <th>End At:</th>
+          <th>Repeat for:</th>
+        </tr>
+      </thead>
+      <tbody>
+        {NewRow("Sunday")}
+        {NewRow("Monday")}
+        {NewRow("Tuesday")}
+        {NewRow("Wednesday")}
+        {NewRow("Thursday")}
+        {NewRow("Friday")}
+        {NewRow("Saturday")}
+      </tbody>
+    </table>
+  );
+}
+
+export function NewRow(name: string) {
```

Comment:
Bit of advice for the typescript components in React. This is definitely some strange syntax so not expecting you to know it. You can create an Interface for the expected props to a component like in my code example below. 

I also think a better name might be CalendarSelectRow or DaySelectRow or something like that. NewRow is a little vague. 

Same goes for name as a parameter. I think dayOfWeek would be better in this case. 


```
interface CalendarSelectRowProps {
    dayOfWeek: string;
}

export const CalendarSelectRow: React.FC<CalendarSelectRowProps> = ({ dayOfWeek }) => {

}
```